tests/unit/lib/charms/sdcore_nms_k8s/v0/test_fiveg_core_gnb_provider_interface.py

```diff
@@ -0,0 +1,168 @@
+TEST_TAC_INVALID = 0
+TEST_MCC = "001"
+TEST_MNC = "01"
+TEST_SST = 1
+TEST_SD = 2
+
+
+class TestFivegCoreGnbProviderCharm:
+    @pytest.fixture(autouse=True)
+    def context(self):
+        self.ctx = scenario.Context(
+            charm_type=DummyFivegCoreGnbProviderCharm,
+            meta={
+                "name": "fiveg-core-gnb-provider",
+                "provides": {
+                    "fiveg_core_gnb": {
+                        "interface": "fiveg_core_gnb",
+                    }
+                },
+            },
+            actions={
+                "publish-gnb-config": {
+                    "params": {
+                        "tac": {
+                            "type": "string",
+                        },
+                        "plmns": {
+                            "type": "string",
+                        },
+                    },
+                },
+                "publish-gnb-config-wrong-data": {
+                    "params": {
+                        "tac": {
+                            "type": "string",
+                        },
+                        "plmns": {
+                            "type": "string",
+                        },
+                    },
+                },
+            },
+        )
+
+    def test_given_fiveg_core_gnb_relation_when_publish_gnb_config_valid_tac_then_data_is_in_application_databag(  # noqa: E501
+        self,
+    ):
+        fiveg_core_gnb_relation = scenario.Relation(
+            endpoint="fiveg_core_gnb",
+        )
+        state_in = scenario.State(
+            leader=True,
+            relations={fiveg_core_gnb_relation},
+        )
+
+        plmns = [PLMNConfig(mcc=TEST_MCC, mnc=TEST_MNC, sst=TEST_SST, sd=TEST_SD)]
+        params = {
+            "tac": str(TEST_TAC_VALID),
+            "plmns": json.dumps([plmn.asdict() for plmn in plmns])
+        }
+
+        state_out = self.ctx.run(self.ctx.on.action("publish-gnb-config", params=params),
+                                 state_in)
+        assert (
+            state_out.get_relation(fiveg_core_gnb_relation.id).local_app_data["tac"]
+            == str(TEST_TAC_VALID)
+        )
+        rel_plmns = state_out.get_relation(fiveg_core_gnb_relation.id).local_app_data["plmns"]
+        assert plmns == [PLMNConfig(**data) for data in json.loads(rel_plmns)]
+
+    def test_given_fiveg_core_gnb_relation_when_publish_gnb_config_invalid_tac_then_exception_is_raised(  # noqa: E501
+        self,
+    ):
+        fiveg_core_gnb_relation = scenario.Relation(
+            endpoint="fiveg_core_gnb",
+        )
+        state_in = scenario.State(
+            leader=True,
+            relations={fiveg_core_gnb_relation},
+        )
+
+        plmns = [PLMNConfig(mcc=TEST_MCC, mnc=TEST_MNC, sst=TEST_SST, sd=TEST_SD)]
+        params = {
+            "tac": str(TEST_TAC_INVALID),
+            "plmns": json.dumps([plmn.asdict() for plmn in plmns])
+        }
+
+        with pytest.raises(Exception):
```

Comment:
```suggestion
        with pytest.raises(Exception) as e:
```
Then we can check the error description.

assert "Invalid relation data" in str(e.value)